pkg/config/config.go

```diff
@@ -389,6 +390,9 @@ type StorageConfig interface {
 	SigningKey() SecureString
 }
 
+// CustomizationConfig unstructured configuration for customizations
+type CustomizationConfig map[string]interface{}
```

Comment:
```suggestion
// CustomizationConfig holds unstructured configuration for customizations.
type CustomizationConfig struct {
    Values map[string]interface{}
}
```

Don't hold a primitive type in a top-level message or config object (any DTO, really).  The day will come when you want to add an important field.  For instance, you might want a field `Enabled bool`.  If all you have is a map, you now need to encode the type-safety assumption "the case `cc["enabled"].(bool)` always succeeds" in documentation.  If you have a sub-field of a struct, this problem goes away.

Reply:
AFAIR we explicitly decided to leave this open in the design until we needed it.  Now we need it.  Please don't skip doing it properly.
The current PR is just one of the options discussed, I think we had better options.